src/components/downloadable-files/index.tsx

```diff
@@ -23,14 +27,50 @@ function DownloadableFiles({
    */
   layout?: LayoutType
 }) {
+  const submissionPDFFileName = React.useMemo(() => {
+    const customFileName =
+      formSubmissionResult.definition.postSubmissionReceipt?.allowPDFDownload
+        ?.pdfFileName
+    if (!customFileName) {
+      return 'Submission'
+    }
+
+    return localisationService.replaceInjectablesWithSubmissionValues(
+      customFileName,
+      {
+        previousApprovalId:
+          formSubmissionResult.previousFormSubmissionApprovalId,
+        form: formSubmissionResult.definition,
+        submission: formSubmissionResult.submission,
+        submissionId: formSubmissionResult.submissionId || '',
+        submissionTimestamp: formSubmissionResult.submissionTimestamp || '',
+        externalId: formSubmissionResult.externalId || undefined,
+        userProfile: authService.getUserProfile() || undefined,
+        task: formSubmissionResult.taskCompletion?.task,
+        taskGroup: formSubmissionResult.taskCompletion?.taskGroup,
+        taskGroupInstance:
+          formSubmissionResult.taskCompletion?.taskGroupInstance,
+      },
+    ).text
+  }, [
+    formSubmissionResult.definition.postSubmissionReceipt?.allowPDFDownload
+      ?.pdfFileName,
+    formSubmissionResult.previousFormSubmissionApprovalId,
+    formSubmissionResult.submission,
+    formSubmissionResult.submissionId,
+    formSubmissionResult.submissionTimestamp,
+    formSubmissionResult.externalId,
+    formSubmissionResult.taskCompletion?.task,
+    formSubmissionResult.taskCompletion?.taskGroup,
+    formSubmissionResult.taskCompletion?.taskGroupInstance,
```

Comment:
Probably cleaner just to put `formSubmissionResult` in here. The whole object changes at the same time anyway.
Also, linter is complaining.